packages/neon/neon/lib/src/utils/dialog.dart

```diff
@@ -0,0 +1,113 @@
+    ) ??
+    false;
+
+/// Displays a [NeonRenameDialog] with the given [title] and [initialValue].
+///
+/// Returns a future with the new name of name.
+Future<String?> showRenameDialog({
+  required final BuildContext context,
+  required final String title,
+  required final String initialValue,
+}) async =>
+    showAdaptiveDialog<String?>(
+      context: context,
+      builder: (final context) => NeonRenameDialog(
+        title: title,
+        value: initialValue,
+      ),
+    );
+
+/// Displays a [NeonErrorDialog] with the given [message].
+Future<void> showErrorDialog({
+  required final BuildContext context,
+  required final String message,
+  final String? title,
+}) =>
+    showAdaptiveDialog<void>(
+      context: context,
+      builder: (final context) => NeonErrorDialog(content: message),
+    );
+
+/// Displays a [NeonDialog] with the given [title] informing the user that a
+/// feature is not implemented yet.
+Future<void> showUnimplementedDialog({
+  required final BuildContext context,
+  required final String title,
+}) =>
+    showAdaptiveDialog<void>(
+      context: context,
+      builder: (final context) => NeonDialog(
+        automaticallyShowCancel: false,
+        title: Text(title),
+        actions: [
+          NeonDialogAction(
+            isDestructiveAction: true,
+            onPressed: () {
+              Navigator.of(context).pop();
+            },
+            child: Text(
+              NeonLocalizations.of(context).actionClose,
+              textAlign: TextAlign.end,
+            ),
+          ),
+        ],
+      ),
+    );
+
+/// Displays a [NeonCreateFolderDialog] for creating a new folder.
+///
+/// Returns a future with the folder name split by `/`.
+Future<String?> showFolderCreateDialog({
+  required final BuildContext context,
+}) =>
+    showAdaptiveDialog<String>(
+      context: context,
+      builder: (final context) => const NeonCreateFolderDialog(),
+    );
+
+/// Displays a [NeonConfirmationDialog] for deleting a folder.
+///
+/// Returns a future whether the action has been accepted.
+Future<bool> showFolderDeleteDialog({
+  required final BuildContext context,
+  required final String folderName,
+}) async {
+  final content = NeonLocalizations.of(context).folderDeleteConfirm(folderName);
+
+  final result = await showAdaptiveDialog<bool>(
+    context: context,
+    builder: (final context) => NeonConfirmationDialog(
+      title: NeonLocalizations.of(context).actionDeleteTitle,
+      content: Text(content),
+    ),
+  );
+
+  return result ?? false;
+}
+
+/// Displays a [NeonRenameDialog] for renaming a folder.
+///
+/// Returns a future with the new name of name.
+Future<String?> showFolderRenameDialog({
+  required final BuildContext context,
+  required final String folderName,
+}) async =>
+    showRenameDialog(
+      context: context,
+      title: NeonLocalizations.of(context).folderRename,
+      initialValue: folderName,
+    );
```

Comment:
IMO these do not belong here, all the other ones can stay.

Reply:
Yeah I get that and I agree with you, but where do you draw the line? If it is the exact same code snippet?